app/components/baseComponents/SmDropdown/SmDropdown.js

```diff
@@ -0,0 +1,315 @@
+        };
+      }
+
+      if (val.disabled) {
+        return {
+          borderColor: Colors.borderGray,
+          backgroundColor: Colors.white,
+          cursor: 'default'
+        };
+      }
+      return { borderColor: Colors.borderGray, backgroundColor: Colors.white };
+    };
+
+    const textLabelDisabledStyle = (val?: DropdownEntry) => {
+      if (disabled || (!!val && val.disabled)) {
+        return styles.disabledLabelText;
+      }
+      return {};
+    };
+
+    const dropdownEntryElem = (val: DropdownEntry, isFirst?: boolean) => (
+      <div style={{ ...styles.entryOuter, ...outerStyle(!!isFirst) }}>
+        <div style={{ ...styles.labelWrapper, ...labelWrapperStyle(val, !!isFirst) }}>
+          <span style={{ ...styles.labelText, ...textLabelDisabledStyle(val) }}>{val.label}</span>
+        </div>
+      </div>
+    );
+
+    const rootDisabledStyle = () => {
+      if (disabled) {
+        return styles.disabled;
+      }
+      return {};
+    };
+
+    const headerStyle = () => {
+      if (hoveredHeader) {
+        return { backgroundColor: `rgba(${Colors.greenRgb},0.1)` };
+      }
+      return {};
+    };
+
+    return (
+      <div style={{ ...styles.root, ...rootDisabledStyle() }}>
+        <div style={{ ...styles.header, ...headerStyle() }} onClick={this.handleToggle} onMouseEnter={this.handleHeaderEnter} onMouseLeave={this.handleHeaderLeave}>
+          <div style={styles.iconWrapper}>
+            <img src={isOpen || hoveredHeader ? openDDIcon : openDDIconDisabled} style={styles.icon} alt="Icon missing" />
+          </div>
+          <span style={{ ...styles.labelText, ...textLabelDisabledStyle() }}>{selectedEntry ? selectedEntry.label : placeholder || DEFAULT_PLACEHOLDER}</span>
+        </div>
+        {isOpen && (
+          <div style={styles.itemsWrapper}>
+            {data.map((val) => (
+              <div
+                key={val.id}
+                onClick={(e: Event) => {
+                  e.preventDefault();
+                  e.stopPropagation();
+                  if (!(disabled || val.disabled)) {
+                    this.handleSelect(val.id);
+                  }
+                }}
+              >
+                <div onMouseEnter={this.handleMouseEnter.bind(this, val.id)} onMouseLeave={this.handleMouseLeave}>
+                  {dropdownEntryElem(val)}
+                </div>
+              </div>
+            ))}
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  componentDidUpdate() {
+    const { isOpen } = this.state;
+    setTimeout(() => {
+      if (isOpen) {
+        window.addEventListener('click', this.closeDropdown);
+      } else {
+        window.removeEventListener('click', this.closeDropdown);
+      }
+    }, 0);
+  }
+
+  handleMouseEnter = (id: number | string) => {
+    this.setState({ hovered: id });
+  };
+
+  handleMouseLeave = () => {
+    this.setState({ hovered: undefined });
+  };
+
+  handleHeaderEnter = () => {
```

Comment:
handleHeaderEnter = () => {
    const { disabled } = this.props;
    if (!disabled) {
      this.setState({ hoveredHeader: true });
    }
  };

Reply:
done